sql/types/json_value.go

```diff
@@ -27,92 +27,63 @@ import (
-	Set(ctx *sql.Context, path string, val JSONValue) (MutableJSONValue, bool, error)
+	Set(ctx *sql.Context, path string, val sql.JSONWrapper) (MutableJSONV, bool, error)
 	// Replace the value at the given path with the new value. If the path does not exist, no modification is made.
-	Replace(ctx *sql.Context, path string, val JSONValue) (MutableJSONValue, bool, error)
+	Replace(ctx *sql.Context, path string, val sql.JSONWrapper) (MutableJSONV, bool, error)
 	// ArrayInsert inserts into the array object referenced by the given path. If the path does not exist, no modification is made.
-	ArrayInsert(ctx *sql.Context, path string, val JSONValue) (MutableJSONValue, bool, error)
+	ArrayInsert(ctx *sql.Context, path string, val sql.JSONWrapper) (MutableJSONV, bool, error)
 	// ArrayAppend appends to an  array object referenced by the given path. If the path does not exist, no modification is made,
 	// or if the path exists and is not an array, the element will be converted into an array and the element will be
 	// appended to it.
-	ArrayAppend(ctx *sql.Context, path string, val JSONValue) (MutableJSONValue, bool, error)
+	ArrayAppend(ctx *sql.Context, path string, val sql.JSONWrapper) (MutableJSONV, bool, error)
 }
 
 type JSONDocument struct {
 	Val interface{}
 }
 
-var _ JSONValue = JSONDocument{}
+var _ sql.JSONWrapper = JSONDocument{}
 
-func (doc JSONDocument) Unmarshall(_ *sql.Context) (JSONDocument, error) {
-	return doc, nil
+func (doc JSONDocument) ToInterface() interface{} {
+	return doc.Val
 }
 
-func (doc JSONDocument) Compare(ctx *sql.Context, v JSONValue) (int, error) {
-	other, err := v.Unmarshall(ctx)
-	if err != nil {
-		return 0, err
-	}
-	return compareJSON(doc.Val, other.Val)
+func (doc JSONDocument) Compare(other sql.JSONWrapper) (int, error) {
+	return CompareJSON(doc.Val, other.ToInterface())
 }
 
-func (doc JSONDocument) ToString(_ *sql.Context) (string, error) {
+func (doc JSONDocument) ToString() (string, error) {
 	return marshalToMySqlString(doc.Val)
 }
 
-var _ SearchableJSONValue = JSONDocument{}
-var _ MutableJSONValue = JSONDocument{}
+var _ sql.JSONWrapper = JSONDocument{}
+var _ MutableJSONV = JSONDocument{}
 
 // Contains returns nil in case of a nil value for either the doc.Val or candidate. Otherwise
 // it returns a bool
-func (doc JSONDocument) Contains(ctx *sql.Context, candidate JSONValue) (val interface{}, err error) {
-	other, err := candidate.Unmarshall(ctx)
-	if err != nil {
-		return false, err
-	}
-	return containsJSON(doc.Val, other.Val)
+func (doc JSONDocument) Contains(candidate sql.JSONWrapper) (val interface{}, err error) {
+	return ContainsJSON(doc.Val, candidate.ToInterface())
+}
+
+func (doc JSONDocument) Extract(path string) (sql.JSONWrapper, error) {
+	return LookupJsonValue(doc, path)
 }
 
-func (doc JSONDocument) Extract(ctx *sql.Context, path string) (JSONValue, error) {
+func LookupJsonValue(j sql.JSONWrapper, path string) (sql.JSONWrapper, error) {
```

Comment:
(minor) Looks like other references to json use `JSON` in all caps. Would be nice to be consistent.